libraries/exoplayer/src/main/java/androidx/media3/exoplayer/AudioFocusManager.java

```diff
@@ -455,6 +453,13 @@ private void executePlayerCommand(@PlayerCommand int playerCommand) {
     }
   }
 
+  private AudioManager getAudioManager() {
+    if (audioManager == null) {
```

Comment:
NIt: you can use Guava's `Suppliers.memoize` for this, to avoid manually needing to check for null, so change `private final AudioManager audioManager;` above to `private final Supplier<AudioManager> audioManager;`, change the instantiation to:
```
this.audioManager =
        Suppliers.memoize(() -> checkNotNull(
            (AudioManager) context.getApplicationContext().getSystemService(Context.AUDIO_SERVICE)));
```

And then change the usage sites from `audioManager` to `audioManager.get()`

Maybe change the field name to `audioManagerSupplier` too - up to you.

Reply:
Good suggestion. Will change accordingly